dotcom-rendering/src/web/components/SignInGate/displayRule.ts

```diff
@@ -30,6 +34,20 @@ export const isIOS9 = (): boolean => {
 	return appleDevice && os;
 };
 
+// determine if the browser is a mobile device if it supports touch points, with user agent fallback
+// https://developer.mozilla.org/en-US/docs/Web/HTTP/Browser_detection_using_the_user_agent
+export const isMobile = (): boolean => {
+	if (navigator.maxTouchPoints) {
+		return navigator.maxTouchPoints > 0;
+	} else {
+		const ua = navigator.userAgent;
+		return (
+			/\b(BlackBerry|webOS|iPhone|IEMobile)\b/i.test(ua) ||
+			/\b(Android|Windows Phone|iPad|iPod)\b/i.test(ua)
+		);
+	}
+};
+
```

Comment:
Can this be moved to a separate PR?

What’s the reason for introducing this `isMobile` check? I don’t think we tend to vary the behaviour of the website based on the reader’s device type.

Reply:
Its a requirement for the new long AB test we need to run - more info[ here ](https://docs.google.com/document/d/10sa-pzxABeOQ6_gXzqTpdh5AKoiK_t69BU2xsLTeIc8/edit#heading=h.2pyb60zfmnx6)

I'll also be adding a display rule to check region, specifically EU for this test. Will put that and `isMobile` into separate PRs